crates/rpc-types/src/eth/admin.rs

```diff
@@ -0,0 +1,410 @@
+//! Types for the admin api
+use alloy_genesis::ChainConfig;
+use alloy_primitives::{B256, U256};
+use alloy_serde::json_u256::deserialize_json_u256;
+use enr::{k256::ecdsa::SigningKey, Enr};
+use serde::{Deserialize, Serialize};
+use std::net::{IpAddr, SocketAddr};
+
+/// This includes general information about a running node, spanning networking and protocol
+/// details.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct NodeInfo {
+    /// The node's private key.
+    pub id: B256,
+
+    /// The node's user agent, containing a client name, version, OS, and other metadata.
+    pub name: String,
+
+    /// The enode URL of the connected node.
+    pub enode: String,
+
+    /// The [ENR](https://eips.ethereum.org/EIPS/eip-778) of the running client.
+    pub enr: Enr<SigningKey>,
```

Comment:
I think, if we replace this with String, we can get rid of the enr dep entirely

Reply:
Done.